### PR TITLE
Improving the collect logs to show the number of available ephemeral ports and the hns network layer info

### PR DESCRIPTION
**Pull request by JocelynBerrendonner**
Improving the collect logs to show the number of available ephemeral ports and the hns network layer info
- Added a function to count the available port ranges which have room for allocating 64 ports chunks
- Improved the Get-HNSNetwork logic to show detailed information

Tested:
- Ran the script on a server SKU computer and verified the information was accurate.

### CODE REVIEW

File: Kubernetes/windows/debug/collectlogs.ps1 [hunk 1]
```diff
@@ -34,7 +34,7 @@ pushd
 cd $outDir
 
 Get-HnsNetwork | Select Name, Type, Id, AddressPrefix > network.txt
-Get-hnsnetwork | Convertto-json -Depth 20 >> network.txt
+Get-hnsnetwork | % { Get-HnsNetwork -Id $_.ID -Detailed } | Convertto-json -Depth 20 >> network.txt
```
Comment (madhanrm) [inline Kubernetes/windows/debug/collectlogs.ps1]: can you write it to networkdetailed.txt

File: Kubernetes/windows/debug/collectlogs.ps1 [hunk 2]
```diff
@@ -65,8 +65,54 @@ if ($res)
...
+
+if ($availableRangesFor64PortChunks -le 0) {
+    echo "ERROR: Running out of ephemeral ports. The ephemeral ports range doesn't have enough resources to allow allocating 64 contiguous TCP ports." > reservedports.txt
+} else {
+    echo "The ephemeral port range still has room for making up to $availableRangesFor64PortChunks allocations of 64 contiguous TCP ports" > reservedports.txt
+}
+
+netsh int ipv4 sh excludedportrange TCP >> reservedports.txt
 netsh int ipv4 sh excludedportrange UDP >> reservedports.txt
+netsh int ipv4 sh dynamicportrange TCP >> reservedports.txt
```
Comment (madhanrm) [inline Kubernetes/windows/debug/collectlogs.ps1]: can you write them to excludedportrange.txt and dynamicportrange.txt

File: Kubernetes/windows/debug/collectlogs.ps1 (hunk 1 above)
Comment (JocelynBerrendonner) [reply Kubernetes/windows/debug/collectlogs.ps1]: Done in latest iteration

File: Kubernetes/windows/debug/collectlogs.ps1 (hunk 2 above)
Comment (JocelynBerrendonner) [reply Kubernetes/windows/debug/collectlogs.ps1]: Done in latest iteration

**Pull request merged**